### PR TITLE
#231 convert plane position sequences to hd.PlanePositionSlideSequence

### PR DESCRIPTION
**Pull request by thomas-albrecht**
This is to fix the Issue #231 that I created myself.

### CODE REVIEW

**Comment by thomas-albrecht** [comment]
Even on the unchanged version that I cloned a lot of tests fail.

File: setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def get_version():
     package_dir={'': 'src'},
     python_requires='>=3.6',
     install_requires=[
-        'pydicom>=2.3.0',
+        'pydicom>=2.3.0,<2.4',
```
Comment (thomas-albrecht) [inline setup.py:54]: `pydicom` 2.4 added some very strict tests on data types, which are not satisfied by `highdicom` calls like [this one](https://github.com/ImagingDataCommons/highdicom/blob/master/src/highdicom/seg/sop.py#L1643):
```
                frame_content_item.DimensionIndexValues = (
                    [segment_number] + index_values
                )
```

File: tests/test_seg.py [hunk 2]
```diff
@@ -2512,6 +2512,31 @@ def test_spatial_positions_not_preserved(self):
         assert po_item.ImageOrientationSlide == list(image_orientation)
         self.check_dimension_index_vals(instance)
 
+    def test_get_plane_positions_of_image_patient(self):
+        seq = DimensionIndexSequence(
+            coordinate_system='PATIENT'
+        )
+        plane_positions = seq.get_plane_positions_of_image(self._ct_multiframe)
+        for position in plane_positions:
+            assert isinstance(position, PlanePositionSequence)
```
Comment (thomas-albrecht) [inline tests/test_seg.py:2521]: This fails with the current version of `src/highdicom/seg/content.py`

**Comment by CPBridge** [comment]
Hi @thomas-albrecht thanks a lot for reporting this bug and implementing the fix with tests, really appreciate the help.

However, can you please remove your change to the setup.py. I am aware of the issue with pydicom 2.4 and will look into it soon. It is a much larger issue that this pull requests and I will fix it in another PR. If your tests pass with pydicom==2.3.1 I am satisfied, don't try and make them pass with 2.4 just yet

**Comment by CPBridge** [comment]
c.f. #233

**Comment by thomas-albrecht** [comment]
Hi @CPBridge, sure I can remove the setup.py thing. 

I did not actually try to make the tests pass with pydicom 2.4. I just specified the version of `pydicom` that `highdicom` currently works with: `pydicom>=2.3.0,<2.4` because 2.4 does not work yet. 

As it is without this change, if you just clone and install it, most tests will fail. It's good to hear though that you are looking into making that work separately.

**Comment by CPBridge** [comment]
Thanks @thomas-albrecht . For reference, there were various issues introduced by pydicom 2.4.0. I currently have two PRs on highdicom waiting to be merged to fix this: #239 and #238. Once those are merged, pipelines on this PR should run correctly.

I'm approving this but may wait to merge it until other things are sorted out

**Review by CPBridge — APPROVED** [review]

**Comment by CPBridge** [comment]
Hi @thomas-albrecht, highdicom's master branch is now in a working state again after the chaos of the pydicom 2.4.0 release. Could you please rebase your branch on master, which should get tests working on this branch again?

Furthermore after running the github actions I noticed that there are some [linter errors](https://github.com/ImagingDataCommons/highdicom/actions/runs/5308927278/jobs/9663504838?pr=232) that I'm hoping you might be able to fix. Thanks!

**Comment by thomas-albrecht** [comment]
Hi @CPBridge, I am a bit embarrassed. I obviously don't know how to use `git rebase`. 
I tried to rebase to the master branch but now my branch contains hundreds of old commits and shows 42 changed files. 

I just did a manual rebase by starting from scratch in the pull request https://github.com/ImagingDataCommons/highdicom/pull/240 and will close this one.

**Pull request closed without merging**